ext/cedar_policy/src/request.rs

```diff
@@ -1,32 +1,43 @@
-use cedar_policy::Request;
-use magnus::{function, method, Error, Module, Object, RModule, Ruby, Value};
+use cedar_policy::{Request, Schema};
+use magnus::{
+    function, method,
+    scan_args::{get_kwargs, scan_args},
+    Error, Module, Object, RModule, Ruby, Value,
+};
 use std::convert::Into;
 
 use crate::{
     context::ContextWrapper,
     entity_uid::{to_euid_value, EntityUidWrapper},
+    error::REQUEST_VALIDATION_ERROR,
+    schema::RSchema,
 };
 
 #[magnus::wrap(class = "CedarPolicy::Request")]
 pub struct RRequest(Request);
 
 impl RRequest {
-    fn new(
-        ruby: &Ruby,
-        principal: EntityUidWrapper,
-        action: EntityUidWrapper,
-        resource: EntityUidWrapper,
-        context: ContextWrapper,
-    ) -> Result<Self, Error> {
+    fn new(ruby: &Ruby, args: &[Value]) -> Result<Self, Error> {
+        let args = scan_args::<_, (), (), (), _, ()>(args)?;
+        let (principal, action, resource, context): (
+            EntityUidWrapper,
+            EntityUidWrapper,
+            EntityUidWrapper,
+            ContextWrapper,
+        ) = args.required;
+        let kw_args = get_kwargs::<_, (), (Option<&RSchema>,), ()>(args.keywords, &[], &["schema"])?;
+        let (schema,) = kw_args.optional;
+        let schema = schema.map(|s| Schema::from(s));
+
         Ok(Self(
             Request::new(
                 principal.into(),
                 action.into(),
                 resource.into(),
                 context.into(),
-                None,
+                schema.as_ref(),
```

Comment:
Does the `schema.into()` can work here?

Reply:
I may be running into the limits of my rust knowledge, but the schema parameter passed to `Request::new` needs to be an `Option<&Schema>`, and I don't think `into()` can make that sort of type (it seems like it is explicitly designed to return **values**, not references). We __can__ replace:

    let schema = schema.map(|s| Schema::from(s))

with:

    let schema: Option<Schema> = schema.map(|s| s.into())

We would still need `as_ref()` to convert `Option<Schema>` to `Option<&Schema>` as required by `Request::new`. 

Is it better to use the explicit type annotation with `into()` than to use `Schema::from()`?